tools/shell/embedded_shell.cpp

```diff
@@ -304,12 +303,53 @@ EmbeddedShell::EmbeddedShell(std::shared_ptr<Database> database, std::shared_ptr
     KU_ASSERT(signal(SIGINT, interruptHandler) != SIG_ERR);
 }
 
-void EmbeddedShell::run() {
-    char* line;
+std::vector<std::unique_ptr<QueryResult>> EmbeddedShell::processInput(std::string input) {
     std::string query;
     std::stringstream ss;
+    std::vector<std::unique_ptr<QueryResult>> queryResults;
+    if (continueLine) {
+        input = std::move(currLine) + std::move(input);
+        currLine = "";
+        continueLine = false;
+    }
+    input = input.erase(input.find_last_not_of(" \t\n\r\f\v") + 1);
+    if (!continueLine && input[0] == ':') {
```

Comment:
Let's leave some comments for these if else statements for future maintainability purpose.